src/main/groovy/com/calgaryscientific/gradle/VeracodeAPI.groovy

```diff
@@ -44,15 +45,29 @@ class VeracodeAPI {
         this.id = id
     }
 
+    String getPreScanResults(String app_id) {
+        UploadAPIWrapper api = uploadAPI()
+        return uploadAPI().getPreScanResults(app_id)
+    }
+
+    String getPreScanResults(String app_id, String build_id) {
+        UploadAPIWrapper api = uploadAPI()
+        return uploadAPI().getPreScanResults(app_id, build_id)
+    }
+
     String uploadFile(String app_id, String filePath) {
         UploadAPIWrapper api = uploadAPI()
         return api.uploadFile(app_id, filePath)
     }
 
     private boolean useAPICredentials() {
-        if (this.username != "" && this.password != "") {
+        if (this.username != null && this.password != null &&
```

Comment:
sounds like you can use groovy truth here .... null and empty string will return false .... so you can just use if (!username and !password) is probably good enough